tests/foreman/ui/test_host.py

```diff
@@ -2150,3 +2151,317 @@ def test_host_status_honors_taxonomies(
+            packages_to_remove=packages_to_remove,
+            manage_by_rex=manage_by_rex_flag,
+            manage_by_customized_rex=manage_by_customized_rex_flag,
+        )
+
+        # Wait till the job launched by the management action is finished
+        job_id = int(module_target_sat.cli.JobInvocation().list()[0]['id'])
+        if install_packages_flag:
+            module_target_sat.wait_for_tasks(
+                search_query=(
+                    f'action: "Install package(s) name ^ ({",".join(packages_to_install)})" and resource_id = {job_id}'
+                ),
+            )
+
+        elif upgrade_packages_flag and (not upgrade_all_packages_flag):
+            module_target_sat.wait_for_tasks(
+                f'action:  "Update package(s) name ^ ({",".join(packages_to_upgrade)})" and resource_id = {job_id}'
+            )
+
+        elif upgrade_all_packages_flag:
+            if manage_by_rex_flag:
+                module_target_sat.wait_for_tasks(
+                    f'action: "Upgrade all packages" and resource_id = {job_id}'
+                )
+            elif manage_by_customized_rex_flag:
+                module_target_sat.wait_for_tasks(
+                    f'action: "Update package(s)" and resource_id = {job_id}'
+                )
+
+        elif remove_packages_flag:
+            module_target_sat.wait_for_tasks(
+                f'action: "Remove packages name ^ ({",".join(packages_to_remove)})" and resource_id = {job_id}'
+            )
+
+        # MAKE ASSERTS AFTER INSTALLING PACKAGES
+        if used_action == 'install':
+            for host in hosts_to_test:
+                # Check that all the wanted packages are installed
+                assert (
+                    host.run(f'rpm -q {" ".join(packages_to_install)}').status == 0
+                ), 'Some of the wanted packages is not installed!'
+
+                # Get versions of installed packages
+                installed_packages = host.run(
+                    f'rpm -qa {" ".join(packages_to_install)}'
+                ).stdout.split('\n')[:-1]
+                installed_packages_version_dict = {}
+                # Create dict containing {'installed_package_name': 'version', ...} for further checks
+                for package in installed_packages:
+                    package_name, version = package.split('-')[0], '-'.join(package.split('-')[1:])
+                    installed_packages_version_dict[package_name] = version[
+                        : version.rfind('.')
+                    ].strip()
+
+                # Check that the latest version of packages is installed
+                for package in packages_to_install:
+                    assert (
+                        installed_packages_version_dict[package]
+                        == tested_hosts_packages_latest_version_dicts[host.hostname][package]
+                    ), f'Package "{package}" is not installed in the latest version!'
```

Comment:
Optional:
if `packages_latest_version_dict` would be a list and would be created as follows:
```python
packages_latest_version_list = []
for item in packages_list:
    parts = item.split()
    # Remove architecture part from package name
    package_name = parts[0].split('.')[0]
    version = parts[1]  # parts look like ['package_name', 'version', 'repo']
    packages_latest_version_list.append({"name": package_name, "version": version})
```

then you could compare it using
```suggestion
                res = host.execute('rpm --query --all --queryformat="%{NAME},%{VERSION},%{RELEASE},%{ARCH}\n"')
                import csv
                installed_packages = [value for value in csv.DictReader(res.stdout.splitlines(), fieldnames={'name', 'version', 'release', 'arch'})]
                
                
```

The reason why I suggest it is that using 
Pick whatever you like from the whole suggestion, but I insist on using `--queryformat`. Doing regular expression magic on `rpm -q`/`rpm -qa` output is a hazard.